battle-service/src/main/java/by/ibondarau/tictactoe/battleservice/controller/dto/BattleResponseDto.java

```diff
@@ -0,0 +1,21 @@
+package by.ibondarau.tictactoe.battleservice.controller.dto;
+
+import org.springframework.boot.context.properties.ConstructorBinding;
+
+import java.time.Instant;
+import java.util.List;
+
+@ConstructorBinding
+public record BattleResponseDto(
```

Comment:
i assume you will have difficulties using ConstructorBinding on dto's , it will not be really easy to provide all required arguments in constructor. also you object creation will look complex because of a lot of arguments in constructor, as result it may lead to unexpected behavior. let say i will add new aatribute between firstPlayerId and secondPlayerId( or even switch order for these 2 parameters) , in result your invocation of constructor will lead to wrong values in dto's.

you may use builder pattern for this to explicitly set variables. or any other creational pattern(builder look preferable).
for mapping between entities and dtos you may use MapStruct.

Reply:
having this, not sure if record is the type which DTO's should have.